### DISCUSSION TITLE
Error without internet connection

### DISCUSSION THREAD

**Discussion by orioninthesky** [michabirklbauer/instagram_json_viewer#21] [Q&A] [answered]
Hi
why the script don't work in a system without an internet connection?
what is the need to have an active connection? all the necessary data should be contained in the backup file prepared by Instagram, right?
Sorry for the noob question but I want to understand it . Thanks

**Comment by michabirklbauer** [comment, marked as answer]
@orioninthesky that is a legit question, thanks for asking!

The answer is that the connection is needed here:
https://github.com/t0xic-m/instagram_json_viewer/blob/e38592eba9a70b7a1bb752f8473dd1f14fecd51d/instaview.py#L908

My script tries to download all profile pictures for people you chat with, so the chat looks nice with the appropriate icons. Since your backup file only has the profile picture for your own account, all other profile pictures have to be looked up online - that's why the connection is needed.

But I will try to add an "offline mode" in the future, thanks for pointing the issue out!

**Reply by orioninthesky** [reply]
I see :)
and I understand it. An ofline option would be interesting 
Thanks
